Feat/store history

park78951 opened the pull request:
## 한 일
- 히스토리 데이터 관리 클래스 구현
- 히스토리 입력 관리 클래스 구현

## 이슈 번호
- #7

## 참고 사항

park78951 committed:
feat: 기본 레이아웃 클래스 추가

park78951 committed:
feat: history 데이터 관리 Storage 클래스 추가

park78951 committed:
feat: 히스토리 추가 기능 부분 구현 중....

fed-gren reviewed (commented):
feat: 히스토리 추가 기능 부분 구현 중....
스터디 진행상황 상 중간에 커밋을 하게 되더라도 어떤 기능을 구현했는지 등에 대해 적어주시면 좋을 것 같아요.

fed-gren commented on src/practice/ts/Components/Store/Add.ts:
```diff
@@ -0,0 +1,45 @@
+import Storage from "./storage";
+
+class Add {
```
Add라는 네이밍이 살짝 애매한거 같아요. 코드 내용을 보면 history를 추가하니까 AddHistory는 어떨까요?
```suggestion
class AddHistory {
```

park78951 replied:
AddHistory가 확실히 명확한 것 같네요. 수정 반영했습니다.

fed-gren commented on src/practice/ts/Components/Store/Add.ts:
```diff
@@ -0,0 +1,45 @@
+class Add {
+  private formElem: HTMLFormElement;
+  private inputElem: HTMLInputElement;
+  private storageContainer: HTMLUListElement;
```
다른 element들은 뒤에 Elem을 붙이셨는데 storageContainer는 없어서 element가 아닌가? 하는 생각이 들 것 같아요.
변수명에서 List라는 의미도 잘 안드러나지 않나 하는 생각도 드네요.

park78951 replied:
Typescript 때문에 확인할 수 있을 거라고 생각했는데, 리뷰보고 변수명으로 확인할 수 있는게 가장 중요한 것 같다는 생각이드네요. 변수명 통일하는게 확실히 좋은 것 같습니다.

fed-gren commented on src/practice/ts/Components/Store/Add.ts:
```diff
@@ -0,0 +1,45 @@
+  private storage: Storage;
+  private id: number;
+
+  constructor(
```
인자가 여러개니까 객체로 받는 것도 고려해보시면 좋을 것 같아요. 순서가 헷갈릴것 같네요.
```suggestion
  constructor({...})
```

park78951 replied:
완전 동의합니다. 수정사항에 반영했습니다.

fed-gren commented on src/practice/ts/Components/Store/Add.ts:
```diff
@@ -0,0 +1,45 @@
+      const userInput = this.inputElem.value;
+      this.id += 1;
+
+      this.storage.setHistoryItem({
```
setHistoryItem 대신 setHistory는 어떨까요? history가 item을 의미해서 item이 불필요해보이네요.

park78951 replied:
history 단일 아이템임을 변수명에 좀더 명확하게 강조하고 싶어서 썻지만 item이란 말이 없어도 이해할 수 있다고 생각이들어서 수정했습니다.

fed-gren commented on src/practice/ts/Components/Store/Add.ts:
```diff
@@ -0,0 +1,45 @@
+    this.id = 0;
+  }
+
+  addUrlInHistory() {
```
이 메서드에서 하는 일은 이벤트 추가니까 어울리는 메서드명으로 수정하면 좋을 것 같습니다.
```suggestion
  attachEvent() {
```

fed-gren commented on src/practice/ts/Components/Store/Storage.ts:
```diff
@@ -0,0 +1,43 @@
+interface IHistroyItem {
```
https://github.com/fed-gren/Typescript/pull/8/files#diff-b0e0b8ea3bd1b2d41ad1a13d1e56adf6d029e21bd8c8a19eb6d3aa537ca01ea1R30
여기와 마찬가지로 Item이라는 단어가 불필요해 보여요.

fed-gren commented on src/practice/ts/Components/Store/Storage.ts:
```diff
@@ -0,0 +1,43 @@
+    url,
+    title,
+  }: {
+    id: number;
```
이 부분도 interface나 type으로 빼도 될 것 같네요.

fed-gren commented on src/practice/ts/Components/Store/index.ts:
```diff
@@ -0,0 +1,34 @@
+  private storeHistoryContainer: HTMLElement;
+  private form: HTMLFormElement;
+  private input: HTMLInputElement;
+  private historyStorage: HTMLUListElement;
```
Add 클래스와 컨벤션이 일치하면 좋을 것 같습니다. typescript 이다 보니까 타입 선언을 보면 알수있어서 꼭 elem 이라는 단어를 표시해야 하나 싶은 생각도 드는데 의견 부탁드립니다~

park78951 replied:
위에서 말씀드렸는데 컨벤션을 맞추는 작업에서 변수명만으로 element임을 좀더 명확하게 말하고 싶어서, elem을 작성하는 것도 괜찮지 않을까 생각해서 이부분은 elem을 붙이는 방향으로 수정해봤습니다. 혹시 불필요하다고 생각이 드신다면 다시 의견 부탁드려요!

park78951 commented:
@fed-gren 사실 빠르게 기능부터 구현하려고 네이밍이나 분리 고려 안하고 빠르게 코드만 짰는데 출근 중에 좋은 리뷰 읽고 도움 많이 됐습니다 ㅋㅋㅋ 주중에 참고해서 다시 클린하게 좀 수정 해야겠네요.

각각 리뷰해주신 건 수정하면서 코멘트 달도록 할게여~!

park78951 committed:
style: component 내부 파일명 명확한 의미를 위한 수정

park78951 committed:
refactor: historyManager 클래스 contructor type 분리

- constructor 인자로 전달되는 argument의 형태를 객체로 변경
- constructor 인자 type 외부 분리
- 변수명 일부 변경

park78951 committed:
refactor: 메서드명 변경

- setHistoryItem => setHistory
- addUrlInHistory => attachEventToAddHistory

park78951 committed:
refactor: HistoryStore inteface 명 변경

park78951 committed:
feat: HistoryList 클래스의 메서드 인자 타입 분리

park78951 committed:
refactor: root class의 변수명 컨벤션 통일을 위해 수정

park78951 committed:
refactor: HisotryManager interface 명 변경

park78951 committed:
feat: root 클래스에서 초기 데이터 각 클래스 constructor로 주입

park78951 committed:
feat: StoreHistory 클래스 Entry Point에서 초기화

park78951 committed:
fix: input 입력 값 렌더링 안되는 문제 수정

park78951 committed:
feat: 입력후 input 입력값 초기화 되도록 코드 수정

park78951 committed:
feat: title 입력 input element와 label 추가

park78951 committed:
feat: 추가된 title 입력 부분 submit 핸들러에 추가

park78951 committed:
fix: input[type="submit"]를 추가해 submit 안되는 문제 해결
